modules/vaos/app/services/vaos/v2/appointments_service.rb

```diff
@@ -735,25 +749,52 @@ def get_avs_link(appt)
         avs_path(data[:sid])
       end
 
+      def get_avs_pdf(appt, binary)
+        cerner_system_id = extract_cerner_identifier(appt)
+
+        return nil if cerner_system_id.nil?
+
+        avs_resp = unified_health_data_service.get_appt_avs(appt_id: cerner_system_id, include_binary: binary)
+
+        return nil if avs_resp.empty? || avs_resp.nil?
+
+        extract_binary_source(avs_resp)
+
+        avs_resp
+      end
+
+      def extract_binary_source(avs_resp)
+        avs_resp.map do |avs|
+          avs['binary'] = avs['binary'].source if avs['binary'].is_a? Regexp
```

Comment:
Note that we should check with Liz why the binary field is returned as a Regexp when she's back. I couldn't spot anything just looking at their code but I suspect some deserialization/serialization quirk.